loom/engine/bindings/loom/lmSQLite.cpp

```diff
@@ -0,0 +1,467 @@
+        if(result != SQLITE_OK)
+        {
+            lmLogError(gSQLiteGroup, "Error calling finalize for database: %s with Result Code: %i", parentDB->getDBName(), result);
+        }
+        return result;        
+    }
+};
+
+
+//---Statement--- external function initialisation
+int __stdcall Statement::stepAsyncBody(void *param)
+{
+    //get the statement
+    Statement *s = (Statement *)param;
+
+
+    //call the internal SQLite step function, then yield the thread
+    int result = sqlite3_step(s->statementHandle); 
+    loom_thread_yield();
+
+    //mutex so we can call the delegate
+    s->stepAsyncDelegateMutex = loom_mutex_create();
+    loom_mutex_lock(s->stepAsyncDelegateMutex);
+
+    //check the result now
+    if(result == SQLITE_ROW)
+    {
+        s->_OnStatementProgressDelegate.invoke();
+    }
+    else if(result == SQLITE_DONE)
+    {
+//HACK: s->stepAsyncLuaState is not working as it can be unstable... 
+//      need another way to pass the Statement through to OnStatementComplete!!!
+lualoom_pushnative<Statement>(s->stepAsyncLuaState, s);
+s->_OnStatementCompleteDelegate.incArgCount();
+s->_OnStatementCompleteDelegate.invoke();
+lua_pop(s->stepAsyncLuaState, 1);
+    }
+    else
+    {
+        if(result == SQLITE_ERROR)
+        {
+            lmLogError(gSQLiteGroup, "Error calling step for database: %s", s->parentDB->getDBName());
+        }
+//TODO: Should we have an additiona _OnStatementErrorDelegate to call here?
+    }
+
+    //we're done now!
+    loom_mutex_unlock(s->stepAsyncDelegateMutex);
+}
+
+
+
+
+
+//---Connection--- external function / variable initialisation
+NativeDelegate Connection::_OnImportCompleteDelegate;
+
+Statement *Connection::prepare(const char *query)
+{
+    Statement *s = new Statement(this);
+
+    //prepare the database with the query provided
+    int res = sqlite3_prepare_v2(dbHandle, query, -1, &s->statementHandle, NULL);
+    if(res != SQLITE_OK)
+    {
+        lmLogError(gSQLiteGroup, "Error preparing the SQLite database: %s with message: %s", getDBName(), getErrorMessage());
+    }
+    return s;
+}
+
+Connection *Connection::open(const char *database, const char *path, int flags)
+{
+    Connection *c;
+
+    //check for valid database name
+    if((database == NULL) || (database[0] == '\0'))
+    {
+        lmLogError(gSQLiteGroup, "Invalid database name specified!");
+        return NULL;
+    }
+
+    //create the connection
+    c = new Connection();
+    c->databaseName = utString(database);
+    
+    //prep the database path
+    if((path == NULL) || (path[0] == '\0'))
+    {
+        //prefix the system writable path in front of the database name
+        c->databaseFullPath = utString(cocos2d::CCFileUtils::sharedFileUtils()->getWriteablePath().c_str());
+    }
+    else
+    {
+        //assume that the user specified a valid system writeable path!!!
+        c->databaseFullPath = utString(path);
+    }
+    c->databaseFullPath += c->databaseName;
```

Comment:
If it contains a / we assume absolute/relative, if none or it fails then try it at system writable.